tests/php/Service/RegisteredMethodManagerTest.php

```diff
@@ -23,6 +25,18 @@ class RegisteredMethodManagerTest extends SapphireTest
         ],
     ];
 
+    public function setUp()
+    {
+        parent::setUp();
+
+        //Ensure that the handler for notifications is Email - which is stubbed with Swift_NullTransport in test state
+        Injector::inst()->load([
+            Email::class . '.MFA.Notification' => [
+                'class' => Email::class
+            ],
+        ]);
```

Comment:
SapphireTest configures a TestMailer implementation by default. Do you need this?

Reply:
Yes, it is possible to configure `Email::class` to be a service other than `Email::class` - which `SapphireTest` will rectify (maybe, I didn't look)...
However through the magic of Injector the dot context is more specific and allows people to override this only in the `MFA.Notification` context, which will in turn override whatever `SapphireTest` configures.

Reply:
Ah good point. And what's the reasoning for defining a sub-service of this? Are you expecting people to want to use a different mailer service for MFA emails?

Reply:
Yeah, potentially. I don't think extra context is a bad thing - one could configure e.g. a custom emailer for most (mfa) things, and an SMS service for notifications only.
It's quite unlikely I'll admit, but nor does the extra context do any harm as the fall back happens regardless (as noted by the service defined in `notifications.yml` added in this same commit).

The only reason that is defined at all (to what is otherwise the default for that class ID) is because we need to support all versions of SS4, so when https://github.com/silverstripe/silverstripe-framework/pull/8634 is finally updated with unit tests this will still be necessary for 4.0 -> 4.5+ (or whenever it actually gets merged).

Reply:
I think it's pretty edge casey to be honest, and the tradeoff is that you have to mock the service manually in these tests. If someone else were running a test that inadvertently triggers this logic then the email handler would still be set. You may consider putting this logic into a TestState implementation so it'd be global, but personally I'd be more in favour of just using the default email implementation since we haven't got any other examples of code yet that deviate from it.

Reply:
To clarify - I'm suggesting you remove the use of sub-service definitions and use the default mailer

Reply:
Removed context.
`Notification` class is now essentially just:
`Email::create()->setTo($to)->setHTMLTemplate($template)->setData($data)->send()`

If someone wants to use a different service they can Inject `Notification` over `Email.mfa`.